src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -865,9 +872,31 @@ private void BeforeQueryStatusForUpgradePackagesConfig(object sender, EventArgs
                 command.Visible = IsSolutionOpen && IsPackagesConfigSelected();
                 command.Enabled = !isConsoleBusy && IsSolutionExistsAndNotDebuggingAndNotBuilding() && HasActiveLoadedSupportedProject;
             });
+        }
+
+        private async Task<bool> IsPackagesConfigBasedProjectAsync()
+        {
+            var dteProject = EnvDTEProjectInfoUtility.GetActiveProject(VsMonitorSelection);
```

Comment:
should you switch to UI thread here or throw if not on UI thread?

Reply:
Yes we can throw if not on UI thread.